### PR TITLE
[Reviewer: Ellie] Memcached schema changes

### PR DESCRIPTION
**Pull request by AlexHockey**
Ellie, please can you review my changes to migrate Ralf to using JSON for it's session store. This should match the design you reviewed. 

I have tested this in UT. I am also doing a manual live test, but this shouldn't prevent you from reviewing. 

You have already reviewed the docs for this (as part of the previous sprout work).

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -247,6 +260,27 @@ int init_options(int argc, char**argv, struct options& options)
...
+      }
+      else if (strcmp(optarg, "json") == 0)
+      {
+        LOG_INFO("Memcached write format set to 'json'");
+        options.memcached_write_format = MemcachedWriteFormat::JSON;
+      }
+      else
+      {
+        LOG_WARNING("Invalid value for memcached-write-format, using '%s'."
+                    "Got '%s', valid vales are 'json' and 'binary'",
```
Comment (eleanor-merry) [inline src/main.cpp]: *values

File: src/main.cpp [hunk 2]
```diff
@@ -247,6 +260,27 @@ int init_options(int argc, char**argv, struct options& options)
...
+                    ((options.memcached_write_format == MemcachedWriteFormat::JSON) ?
```
Comment (eleanor-merry) [inline src/main.cpp]: Can't you just print out what was received?

File: src/sessionstore.cpp [hunk 3]
```diff
@@ -64,9 +88,23 @@ SessionStore::Session* SessionStore::get_session_data(const std::string& call_id
 
   if (status == Store::Status::OK && !data.empty())
   {
+    // Retrieved the data, so deserialize it.
     rc = deserialize_session(data);
```
Comment (eleanor-merry) [inline src/sessionstore.cpp]: rc is a strange name given that it's session data rather than a return code

**Comment by eleanor-merry** [comment]
Looks good, very minor comments inline

File: src/main.cpp (hunk 2 above)
Comment (AlexHockey) [inline src/main.cpp]: I'm not sure I understand. I am printing two things: the setting we are using, and the option that was received. Can you elaborate?

File: src/sessionstore.cpp (hunk 3 above)
Comment (AlexHockey) [inline src/sessionstore.cpp]: Agreed. I existed before I got here, but I will fix up.

File: src/main.cpp (hunk 2 above)
Comment (eleanor-merry) [inline src/main.cpp]: Misread this, sorry

Comment (AlexHockey) [inline src/main.cpp]: np

**Pull request merged**